ChopChop/Views/OnlineRecipes/OnlineRecipeByUserView.swift

```diff
@@ -32,6 +27,30 @@ struct OnlineRecipeByUserView: View {
         .padding([.horizontal], 100)
     }
 
+    var rateRecipeButton: some View {
+        Button(action: {
+            if viewModel.isShowingRating {
+                if viewModel.ownRating != nil {
+                    viewModel.removeRating()
+                }
+            }
+            viewModel.toggleShowRating()
```

Comment:
Just curious why the rating is closed after removing your rating. And I think the second if statement can be attached to the first one.

Reply:
I realised this part of the code could be improved after looking at it again, and I will do so in the next commit. To answer your first question, any changes to the rating re-renders the entire online recipe view, which resets the `isShowingRating` flag to its default `false`, that's why it "closes the rating"